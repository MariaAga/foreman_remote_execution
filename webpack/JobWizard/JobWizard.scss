.job-wizard {
  .pf-c-wizard__main {
    z-index: calc(
      var(--pf-c-wizard__footer--ZIndex) + 1
    ); // So the select box can be shown above the wizard footer
  }

  .pf-c-wizard__main-body {
    max-width: 500px;
    .advanced-fields-title {
      margin-bottom: 10px;
    }
  }
}
