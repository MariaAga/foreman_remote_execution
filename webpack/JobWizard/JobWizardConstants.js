import { foremanUrl } from 'foremanReact/common/helpers';

export const JOB_TEMPLATES = 'JOB_TEMPLATES';
export const JOB_CATEGORIES = 'JOB_CATEGORIES';
export const JOB_TEMPLATE = 'JOB_TEMPLATE';
export const templatesUrl = foremanUrl('/api/v2/job_templates');
