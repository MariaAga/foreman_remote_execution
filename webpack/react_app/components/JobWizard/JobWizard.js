import React, { useState } from 'react';
import { Wizard } from '@patternfly/react-core';
import { translate as __ } from 'foremanReact/common/I18n';
import CategoryAndTemplate from './steps/CategoryAndTemplate/';
import AdvancedFields from './steps/AdvancedFields/';

export const JobWizard = () => {
  const [jobTemplate, setJobTemplate] = useState(null);
  const [category, setCategory] = useState('');
  const [advancedValue, setAdvancedValue] = useState({
    templateValues: {},
    effectiveUserValue: '',
    timeoutToKill: '',
    password: '',
    keyPassphrase: '',
    effectiveUserPassword: '',
    concurrencyLevel: '',
    timeSpan: '',
  });
  const steps = [
    {
      name: __('Category and template'),
      component: (
        <CategoryAndTemplate
          jobTemplate={jobTemplate}
          setJobTemplate={setJobTemplate}
          category={category}
          setCategory={setCategory}
        />
      ),
    },
    {
      name: __('Target hosts'),
      component: <p>TargetHosts </p>,
      canJumpTo: !!jobTemplate,
    },
    {
      name: __('Advanced fields'),
      component: (
        <AdvancedFields
          // jobTemplateID={jobTemplate}
          jobTemplateID={197}
          advancedValue={advancedValue}
          setAdvancedValue={setAdvancedValue}
        />
      ),
      // canJumpTo: !!jobTemplate,
    },
    {
      name: __('Schedule'),
      component: <p>Schedule</p>,
      canJumpTo: !!jobTemplate,
    },
    {
      name: __('Review details'),
      component: <p>ReviewDetails</p>,
      nextButtonText: 'Run',
      canJumpTo: !!jobTemplate,
    },
  ];
  const title = __('Run Job');
  return <Wizard navAriaLabel={`${title} steps`} steps={steps} height="70vh" />;
};

export default JobWizard;
