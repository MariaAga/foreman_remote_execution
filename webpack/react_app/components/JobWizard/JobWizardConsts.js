export const JOB_TEMPLATES = 'JOB_TEMPLATES';
export const JOB_CATEGORIES = 'JOB_CATEGORIES';
