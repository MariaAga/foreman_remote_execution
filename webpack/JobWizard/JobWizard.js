import React, { useState, useEffect, useCallback } from 'react';
import { useDispatch } from 'react-redux';
import { Wizard } from '@patternfly/react-core';
import { get } from 'foremanReact/redux/API';
import { translate as __ } from 'foremanReact/common/I18n';
import history from 'foremanReact/history';
import CategoryAndTemplate from './steps/CategoryAndTemplate/';
import { AdvancedFields } from './steps/AdvancedFields/AdvancedFields';
import { JOB_TEMPLATE } from './JobWizardConstants';
import './JobWizard.scss';

export const JobWizard = () => {
  const [jobTemplateID, setJobTemplateID] = useState(null);
  const [category, setCategory] = useState('');
  const [advancedValue, setAdvancedValue] = useState({});
  const dispatch = useDispatch();

  const setDefaults = useCallback(response => {
    const responseJob = response.data;
    setAdvancedValue({
      effectiveUserValue: responseJob.effective_user?.value || '',
      timeoutToKill: responseJob.job_template.execution_timeout_interval || '',
    });
  }, []);
  useEffect(() => {
    if (jobTemplateID) {
      dispatch(
        get({
          key: JOB_TEMPLATE,
          url: `/ui_job_wizard/template/${jobTemplateID}`,
          handleSuccess: setDefaults,
        })
      );
    }
  }, [jobTemplateID, dispatch, setDefaults]);

  const steps = [
    {
      name: __('Category and template'),
      component: (
        <CategoryAndTemplate
          jobTemplate={jobTemplateID}
          setJobTemplate={setJobTemplateID}
          category={category}
          setCategory={setCategory}
        />
      ),
    },
    {
      name: __('Target hosts'),
      component: <p>TargetHosts </p>,
      canJumpTo: !!jobTemplateID,
    },
    {
      name: __('Advanced fields'),
      component: (
        <AdvancedFields
          advancedValue={advancedValue}
          setAdvancedValue={newValue => {
            setAdvancedValue(currentAdvancedValue => ({
              ...currentAdvancedValue,
              ...newValue,
            }));
          }}
          jobTemplateID={jobTemplateID}
        />
      ),
      canJumpTo: !!jobTemplateID,
    },
    {
      name: __('Schedule'),
      component: <p>Schedule</p>,
      canJumpTo: !!jobTemplateID,
    },
    {
      name: __('Review details'),
      component: <p>ReviewDetails</p>,
      nextButtonText: 'Run',
      canJumpTo: !!jobTemplateID,
    },
  ];
  const title = __('Run Job');
  return (
    <Wizard
      onClose={() => history.goBack()}
      navAriaLabel={`${title} steps`}
      steps={steps}
      height="70vh"
      className="job-wizard"
    />
  );
};

export default JobWizard;
