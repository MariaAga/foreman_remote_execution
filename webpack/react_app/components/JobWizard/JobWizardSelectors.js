import {
  selectAPIResponse,
  selectAPIStatus,
} from 'foremanReact/redux/API/APISelectors';

import { JOB_TEMPLATES, JOB_CATEGORIES, JOB_TEMPLATE } from './JobWizardConsts';

export const selectJobTemplatesStatus = state =>
  selectAPIStatus(state, JOB_TEMPLATES);

export const selectJobTemplates = state =>
  selectAPIResponse(state, JOB_TEMPLATES)?.results?.filter(
    template => !template.snippet
  ) || [];

export const selectJobCategories = state =>
  selectAPIResponse(state, JOB_CATEGORIES).job_categories || [];

export const selectJobCategoriesStatus = state =>
  selectAPIStatus(state, JOB_CATEGORIES);

export const selectJobTemplate = state =>
  selectAPIResponse(state, JOB_TEMPLATE) || {};
