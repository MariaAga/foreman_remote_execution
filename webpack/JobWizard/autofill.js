import { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { get } from 'foremanReact/redux/API';
import { HOST_IDS } from './JobWizardConstants';
import './JobWizard.scss';

export const useAutoFill = ({
  fills,
  setSelectedTargets,
  setHostsSearchQuery,
}) => {
  const dispatch = useDispatch();

  useEffect(() => {
    if (Object.keys(fills).length) {
      if (fills['host_ids[]']) {
        dispatch(
          get({
            key: HOST_IDS,
            url: '/api/hosts',
            params: { search: `id = ${fills['host_ids[]'].join(' or id = ')}` },
            handleSuccess: ({ data }) => {
              setSelectedTargets(currentTargets => ({
                ...currentTargets,
                hosts: (data.results || []).map(result => result.name),
              }));
            },
          })
        );
      }
      if (fills.search) {
        setHostsSearchQuery(fills.search);
      }
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
};
