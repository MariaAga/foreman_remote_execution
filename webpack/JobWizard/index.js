import React from 'react';
import PropTypes from 'prop-types';
import URI from 'urijs';
import { Title, Divider } from '@patternfly/react-core';
import { translate as __ } from 'foremanReact/common/I18n';
import PageLayout from 'foremanReact/routes/common/PageLayout/PageLayout';
import { JobWizard } from './JobWizard';

const JobWizardPage = ({ location: { search } }) => {
  const fills = URI.parseQuery(search);
  const title = __('Run job');
  const breadcrumbOptions = {
    breadcrumbItems: [
      { caption: __('Jobs'), url: `/jobs` },
      { caption: title },
    ],
  };
  return (
    <PageLayout
      header={title}
      breadcrumbOptions={breadcrumbOptions}
      searchable={false}
    >
      <React.Fragment>
        <Title headingLevel="h2" size="2xl">
          {title}
        </Title>
        <Divider component="div" />
        <JobWizard fills={fills} />
      </React.Fragment>
    </PageLayout>
  );
};

JobWizardPage.propTypes = {
  location: PropTypes.shape({
    search: PropTypes.string,
  }).isRequired,
};

export default JobWizardPage;
