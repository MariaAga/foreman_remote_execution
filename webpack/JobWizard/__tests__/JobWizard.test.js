import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import * as patternfly from '@patternfly/react-core';
import { testComponentSnapshotsWithFixtures, mount } from '@theforeman/test';
import JobWizardPage from '../index';
import { JobWizard } from '../JobWizard';

jest.spyOn(patternfly, 'Wizard');
patternfly.Wizard.mockImplementation(props => <div>{props.navAriaLabel}</div>);
const mockStore = configureMockStore();
const store = mockStore({});
const fixtures = {
  'renders ': {},
};
describe('JobWizardPage rendering', () =>
  testComponentSnapshotsWithFixtures(JobWizardPage, fixtures));

describe('JobWizard rendering', () => {
  describe('render', () => {
    const component = mount(
      <Provider store={store}>
        <JobWizard />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
  describe('update advanced fields', () => {});
});
