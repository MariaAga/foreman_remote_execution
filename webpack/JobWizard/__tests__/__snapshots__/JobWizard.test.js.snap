// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <JobWizard>
    <mockConstructor
      className="job-wizard"
      height="70vh"
      navAriaLabel="Run Job steps"
      onClose={[Function]}
      steps={
        Array [
          Object {
            "component": <ConnectedCategoryAndTemplate
              category=""
              jobTemplate={null}
              setCategory={[Function]}
              setJobTemplate={[Function]}
            />,
            "name": "Category and template",
          },
          Object {
            "canJumpTo": false,
            "component": <p>
              TargetHosts 
            </p>,
            "name": "Target hosts",
          },
          Object {
            "canJumpTo": false,
            "component": <AdvancedFields
              advancedValue={Object {}}
              jobTemplateID={null}
              setAdvancedValue={[Function]}
            />,
            "name": "Advanced fields",
          },
          Object {
            "canJumpTo": false,
            "component": <p>
              Schedule
            </p>,
            "name": "Schedule",
          },
          Object {
            "canJumpTo": false,
            "component": <p>
              ReviewDetails
            </p>,
            "name": "Review details",
            "nextButtonText": "Run",
          },
        ]
      }
    >
      <div>
        Run Job steps
      </div>
    </mockConstructor>
  </JobWizard>
</Provider>
`;

exports[`JobWizardPage rendering renders  1`] = `
<PageLayout
  breadcrumbOptions={
    Object {
      "breadcrumbItems": Array [
        Object {
          "caption": "Jobs",
          "url": "/jobs",
        },
        Object {
          "caption": "Run job",
        },
      ],
    }
  }
  header="Run job"
  searchable={false}
>
  <Title
    headingLevel="h2"
    size="2xl"
  >
    Run job
  </Title>
  <Divider
    component="div"
  />
  <JobWizard />
</PageLayout>
`;
